### PR TITLE
Increase the speed of typing

### PR DESCRIPTION
**Pull request by eglaysher**
A large part of the delay in typing was spent in the kernel where it was scanning the +wa cache because the giant type nouns were reconstructed each time; the VM underneath couldn't do a simple pointer check. Now we use the caching version of +slot in +slur.

In addition, we also were using the compiler in a silly way in +ap-find in Gall, where we were generating arm names and seeing if they existed. Each time we tried to fire that arm. This was 10% of runtime. We cache this value now.

### CODE REVIEW

**Comment by eglaysher** [comment]
(When reviewing, could you patch this in locally to see if I'm just hallucinating the speed increase?)

**Comment by joemfb** [comment]
I tested this locally and typing feels incrementally faster. (FWIW, it had already gotten significantly faster with the %ford hint fix.)

Timing the test suite with `:time _* +test /` didn't show a consistent difference, but that's probably not a great example. I added some timing printfs to `:test [%cores /]` and saw improvements, but the results were inconsistent from one run to the next. I'm not sure how to get actual comparable numbers for event timings.

**Comment by eglaysher** [comment]
I'm going to wait until Monday to commit this, just to make sure everyone who wants to comment sees it first.

You might also want to combine testing this with urbit/urbit#1073, since the allocator also was a big source of latency inside arvo between card dispatch.

**Comment by joemfb** [comment]
That's definitely the fastest combination. But I'm still looking for an appropriate and consistent timing test to compare across these branches.

**Comment by ohAitch** [comment]
How granular are the printf timing annotations under -DGHETTO, and could
this be combined with |verb to compare the bootup sequence on a per-card
basis?

**Comment by joemfb** [comment]
I added `#define GHETTO` to `all.h` and booted a fake-zod in three different configurations: 1) release-candidate Arvo and vere, 2) release-candidate vere with this branch, and 3) urbit/urbit#1073 with this branch. Output (and a `diff3` of the files is attached.

- [time1.txt](https://github.com/urbit/arvo/files/2613702/time1.txt)
- [time2.txt](https://github.com/urbit/arvo/files/2613703/time2.txt)
- [time3.txt](https://github.com/urbit/arvo/files/2613704/time3.txt)
- [time-compare.txt](https://github.com/urbit/arvo/files/2613701/time-compare.txt)

TLDR, number 3 is the fastest, but not for everything. This still isn't an idea test (it'd be better to have a greater variety of events and more granular timing), but it's interesting all the same.

**Comment by ohAitch** [comment]
Can you +solid a lac=| pill? This seems to be on UNIX event granularity and
not |verb granularity.

(Also several runs under each might be useful to get an idea of variance on
any given vere, but idk if those are easy to summarize concisely)

**Comment by joemfb** [comment]
Sure. "before" is release-candidate vere and arvo, "after" is urbit/urbit#1073 and this branch, and "compare" is a diff:

- [time-verb-before.txt](https://github.com/urbit/arvo/files/2613921/time-verb-before.txt)
- [time-verb-after.txt](https://github.com/urbit/arvo/files/2613920/time-verb-after.txt)
- [time-verb-compare.txt](https://github.com/urbit/arvo/files/2613922/time-verb-compare.txt)

**Comment by ohAitch** [comment]
A `--word-diff` might be easier to parse here: [time-verb-compare-word.txt](https://github.com/urbit/arvo/files/2613970/time-verb-compare-word.txt)
This does look like improvements across the board!

Possibly-final fiddling request: it looks like `DGHETTO` [consolidates timings <10ms](https://github.com/urbit/urbit/blob/7672028/noun/trace.c#L55)? Which I can see being useful in debugging(esp if printing takes ~10ms), but makes the relevant prints harder to compare directly, due to minor fluctuations changing which of the lines end up bearing the full timestamps.

**Comment by ohAitch** [comment]
I suppose an adjacent possibility is measuring from boot time, or the beginning of the event, or sth: this can be done in post-processing obviously, but might be easier to eyeball changes in. (somebody with a stats major [help us budget](https://twitter.com/dril/status/384408932061417472?lang=en) our compute)

**Comment by jtobin** [comment]
> somebody with a stats major help us budget our compute
